### ISSUE TITLE
Warning log: set ClimateEntityFeature

### ISSUE THREAD

**Issue by sudomax** [libdyson-wg/ha-dyson#141]
Got the following log:

```
Logger: homeassistant.components.climate
Source: components/climate/__init__.py:361
integration: Climate (documentation, issues)
First occurred: 8:12:08 AM (1 occurrences)
Last logged: 8:12:08 AM

Entity None (<class 'custom_components.dyson_local.climate.DysonPureHotCoolEntity'>) implements HVACMode(s): off, cool, heat and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues
```

**Comment by dotvezz** [comment]
Thanks @sudomax! I'd tried to fix this with a previous update but it didn't work. Are you on a Home Assistant beta, or is this log showing on mainline Home Assistant now?

**Comment by sudomax** [comment]
Mainline. 2024.3 exactly.
Let me know if you need anything else.

**Comment by sayam93** [comment]
I am running HomeAssistant in a docker container on my Raspberry Pi 4B (software and kernels all updated to the latest stable releases) and got the same.

2024-03-18 16:45:45.575 WARNING (MainThread) [homeassistant.components.climate] Entity None (<class 'custom_components.dyson_local.climate.DysonPureHotCoolEntity'>) implements HVACMode(s): off, cool, heat and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues

HA Core: 2024.3.0
HA Frontend: 20240306.0
Docker version 25.0.4, build 1a576c5

**Comment by AnthonyBe** [comment]
I'm still seeing this warning logged with HA 2024.4.4 and integration v1.3.11:

`2024-04-26 10:42:20.363 WARNING (MainThread) [homeassistant.components.climate] Entity None (<class 'custom_components.dyson_local.climate.DysonPureHotCoolLinkEntity'>) implements HVACMode(s): off, cool, heat and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues`

My device is a Dyson Pure Hot+Cool Link fan.

Let me know if there is any other info I can provide to assist investigation!

**Comment by J4yDubsHome** [comment]
Seeing same warning.

Core 2024.5.1
Supervisor 2024.04.4
Operating System 12.2
Frontend 20240501.0

v1.4.0-rc4

`Logger: homeassistant.components.climate
Source: components/climate/__init__.py:363
integration: Climate (documentation, issues)
First occurred: 7:21:45 PM (1 occurrences)
Last logged: 7:21:45 PM

Entity None (<class 'custom_components.dyson_local.climate.DysonPureHotCoolEntity'>) implements HVACMode(s): off, cool, heat and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues`

**Comment by J4yDubsHome** [comment]
Seeing same warning.

Core 2024.5.1
Supervisor 2024.04.4
Operating System 12.2
Frontend 20240501.0

[v1.4.0-rc4](https://github.com/libdyson-wg/ha-dyson/releases/tag/v1.4.0-rc4)

`Logger: homeassistant.components.climate
Source: components/climate/__init__.py:363
integration: Climate (documentation, issues)
First occurred: 7:21:45 PM (1 occurrences)
Last logged: 7:21:45 PM

Entity None (<class 'custom_components.dyson_local.climate.DysonPureHotCoolEntity'>) implements HVACMode(s): off, cool, heat and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues`

**Comment by GentlemanJK** [comment]
Another one for you:

Core 2024.6.1
Supervisor 2024.06.0
Operating System 12.3
Frontend 20240605.0

Running on HA Green

Dyson Pure Hot+Cold Link (Dyson 455) in the USA. Setup using the MyDyson account method. I do know that the climate related functions are limited in the MyDyson app in the USA as well.

Logger: homeassistant.components.climate
Source: components/climate/__init__.py:371
integration: Climate ([documentation](https://www.home-assistant.io/integrations/climate), [issues](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+climate%22))
First occurred: 9:16:12 AM (1 occurrences)
Last logged: 9:16:12 AM

Entity None (<class 'custom_components.dyson_local.climate.DysonPureHotCoolLinkEntity'>) implements HVACMode(s): off, cool, heat and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues

**Comment by nyok92** [comment]
Same issue with HA supervised install : 
```
Enregistreur: homeassistant.components.climate
Source: components/climate/__init__.py:371
intégration: Thermostat ([documentation](https://www.home-assistant.io/integrations/climate), [problèmes](https://github.com/home-assistant/core/issues?q=is%3Aissue+is%3Aopen+label%3A%22integration%3A+climate%22))
S'est produit pour la première fois: 11:55:16 (1 occurrences)
Dernier enregistrement: 11:55:16

Entity None (<class 'custom_components.dyson_local.climate.DysonPureHotCoolEntity'>) implements HVACMode(s): off, cool, heat and therefore implicitly supports the turn_on/turn_off methods without setting the proper ClimateEntityFeature. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues
```

**Comment by dotvezz** [comment]
Hi everyone here! The latest release ([v1.4.0](https://github.com/libdyson-wg/ha-dyson/releases/tag/v1.4.0)) _should_ finally resolve these silly logs. Please feel free to tell me I'm wrong if it doesn't though!

**Comment by sudomax** [comment]
Hello!
I just updated to v1.4.0 and got this warning when rebooting:

```
Logger: homeassistant.components.fan
Source: components/fan/__init__.py:295
integration: Fan (documentation, issues)
First occurred: 8:48:36 PM (2 occurrences)
Last logged: 8:48:36 PM

Entity None (<class 'custom_components.dyson_local.fan.DysonPureCoolEntity'>) does not set FanEntityFeature.TURN_OFF but implements the turn_off method. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues
Entity None (<class 'custom_components.dyson_local.fan.DysonPureCoolEntity'>) does not set FanEntityFeature.TURN_ON but implements the turn_on method. Please create a bug report at https://github.com/libdyson-wg/ha-dyson/issues
```

**Comment by dotvezz** [comment]
@sudomax ironically that's a _different_ new log that looks really similar, but is different. It's brand new as of Home Assistant 2024-8 ([here is a link](https://developers.home-assistant.io/blog/2024/07/19/fan-fanentityfeatures-turn-on_off/) to the Home Assistant blog for more info) and it's for the FanEntity devices. This issue is for ClimateEntity devices.

I was REALLY confused when I saw it too :sweat_smile:. It'll be a quick fix though, tracking in #183.

**Comment by sudomax** [comment]
Thank you! Closing this issue
